### ISSUE TITLE
SC1008 Shell detection fails on "#!/usr/bin/env -S bash"

### ISSUE THREAD

**Issue by Janfel** [koalaman/shellcheck#2143]
#### For bugs
- Rule Id: SC1008
- My shellcheck version: "online" (2021-02-10, master at 15ff87cf80bfcc23fa4a3d0412594e17959dd1b0)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env -S bash
```

#### Here's what shellcheck currently says:
```
Line 1:
#!/usr/bin/env -S bash
^-- SC1008: This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
```

#### Here's what I wanted or expected to see:
No output, `bash` is deduced from the shebang just like with `#!/usr/bin/env bash`.

**Comment by matthewpersicowork** [comment]
What system are you running that `env` takes a `-S` argument?

**Comment by brother** [comment]
GNU coreutils 8.32 does at least. I am on a Debian testing box here.

```
brother ~$ env --help | grep split
  -S, --split-string=S  process and split S into separate arguments;
brother ~$ env --version | head -n1
env (GNU coreutils) 8.32
```
